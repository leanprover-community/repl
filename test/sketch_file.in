{"path": "test/sketch_file.lean", "sorries": "grouped"}

{"tactic": "apply Nat.add_zero", "proofState": 0}

{"tactic": "rw [h_base, h_symm]", "proofState": 2}

{"tactic": "rw [Nat.succ_add, Nat.add_succ, ih]", "proofState": 3}

{"tactic": "apply Nat.zero_add", "proofState": 4}

{"tactic": "(\nintros; rfl)", "proofState": 1}
