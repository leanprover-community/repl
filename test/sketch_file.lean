theorem add_comm_proved_formal_sketch : ∀ n m : Nat, n + m = m + n := by
  intros n m
  induction n with
    | zero =>
      have h_base: 0 + m = m := sorry
      have h_symm: m + 0 = m := sorry
      sorry
    | succ n ih => sorry

theorem add_rfl : ∀ n m : Nat, n + m = n + m := by intros; rfl
theorem add_rfl2 : ∀ n m : Nat, n + m = n + m := by sorry
