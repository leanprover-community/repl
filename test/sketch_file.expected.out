{"sorries":
 [{"proofState": 0,
   "pos": {"line": 1, "column": 70},
   "parentDecl": "add_comm_proved_formal_sketch",
   "goals":
   ["m : Nat\nh_base : 0 + m = m\n⊢ m + 0 = m",
    "m : Nat\nh_base : 0 + m = m\nh_symm : m + 0 = m\n⊢ 0 + m = m + 0",
    "m n : Nat\nih : n + m = m + n\n⊢ n + 1 + m = m + (n + 1)",
    "m : Nat\n⊢ 0 + m = m"],
   "endPos": {"line": 8, "column": 24}},
  {"proofState": 1,
   "pos": {"line": 11, "column": 49},
   "parentDecl": "add_rfl2",
   "goals": ["⊢ ∀ (n m : Nat), n + m = n + m"],
   "endPos": {"line": 11, "column": 57}}],
 "messages":
 [{"severity": "warning",
   "pos": {"line": 1, "column": 8},
   "endPos": {"line": 1, "column": 37},
   "data": "declaration uses 'sorry'"},
  {"severity": "warning",
   "pos": {"line": 11, "column": 8},
   "endPos": {"line": 11, "column": 16},
   "data": "declaration uses 'sorry'"}],
 "env": 0}

{"proofStatus": "Incomplete: open goals remain",
 "proofState": 2,
 "goals":
 ["m : Nat\nh_base : 0 + m = m\nh_symm : m + 0 = m\n⊢ 0 + m = m + 0",
  "m n : Nat\nih : n + m = m + n\n⊢ n + 1 + m = m + (n + 1)",
  "m : Nat\n⊢ 0 + m = m"]}

{"proofStatus": "Incomplete: open goals remain",
 "proofState": 3,
 "goals":
 ["m n : Nat\nih : n + m = m + n\n⊢ n + 1 + m = m + (n + 1)",
  "m : Nat\n⊢ 0 + m = m"]}

{"proofStatus": "Incomplete: open goals remain",
 "proofState": 4,
 "goals": ["m : Nat\n⊢ 0 + m = m"]}

{"proofStatus": "Completed", "proofState": 5, "goals": []}

{"proofStatus": "Completed", "proofState": 6, "goals": []}

